Update fluentd and cleanup Dockerfile

hermitmaster opened the pull request:
Update the fluentd gem to the latest stable version.

Commit:
Update fluentd

Update the fluentd gem to the latest stable version.

dbaldwin-splunk reviewed (commented):
@Gimi Are we ok to commit this change?  Looks minor.

Commit:
Dockerfile cleanup

Removed `set -e` because it literally does nothing in a Dockerfile. Changed the gem install syntax to be less verbose and install all the gems in one go. Switched to installing dumb-init from official Debian repos.

Commit:
Missed a line in cleanup

Remove useless chmod for dum-init.

ghost commented:
I made some additional changes. Details re the changes are in the commit messages. Basically updated fluentd, switched to using dumb-init from Debian repos, and changed the gem install syntax to install all gems at once and cut down on Dockerfile verbosity.

dtregonning commented:
Thanks  for the PR @hermitmaster - we'll incorporate your changes when we update the hec plugin in the next release.
